### PR TITLE
B 21566 int

### PR DESCRIPTION
**Pull request by brianmanley-caci**
## [Agility ticket]
(https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1023571&RoomContext=TeamRoom%3A1007470&concept=TeamRoom)

## Summary
This story is to add the new UBP service item which is replacing these now obsolete service items: ICOUB, IOOUB, IOCUB, NSTUB. We are also adding the new ISLH service item which is replacing: IOCLH, ICOLH, IOOLH, NSTH. The obsolete services are removed from service_params, and re_services. All rows for pricing of the obsolete services in re_intl_prices are now pointed to the new replacment service (UBP or ISLH). The re_service_items table is added to associate a service to a market/shipment type combo.

Additionally we are adding IOSFSC and IDSFSC to re_services and re_service_items.

Is there anything you would like reviewers to give additional scrutiny? Maximum scrutiny on all changes.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

###Database

1. Run the migrations.
2. Verify that IOCUB, ICOUB, IOOUB, NSTUB, IOCLH, ICOLH, IOOLH, NSTH are no longer in the re_services table.
3. Verify that UBP, ISLH, IOSFSC, and IDSFSC are in the re_services_table.
4. Verify that there are rows tied to UBP and ISLH in the re_intl_prices table.
5. Verify that there are rows tied to UBP, ISLH, IOSFSC, and IDSFSC in the re_service_items_table.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Review by begrohmann — APPROVED** [review]

**Review by msaki-caci — APPROVED** [review]

File: pkg/models/re_service_item.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	ServiceId      uuid.UUID       `db:"service_id" rw:"r"`
+	ReService      ReService       `belongs_to:"re_services" fk_id:"service_id" rw:"r"`
+	ShipmentType   MTOShipmentType `db:"shipment_type" rw:"r"`
+	MarketCode     MarketCode      `db:"market_code" rw:"r"`
+	IsAutoApproved bool            `db:"is_auto_approved" rw:"r"`
+	CreatedAt      time.Time       `db:"created_at" rw:"r"`
+	UpdatedAt      time.Time       `db:"updated_at" rw:"r"`
+}
+
+func (g ReServiceItem) TableName() string {
```
Comment (danieljordan-caci) [inline pkg/models/re_service_item.go:20]: Can you change this shortcut to something relevant instead of `g`?

File: pkg/models/re_service.go [hunk 2]
```diff
@@ -121,16 +121,36 @@ const (
...
-	Name      string        `json:"name" db:"name"`
-	CreatedAt time.Time     `json:"created_at" db:"created_at"`
-	UpdatedAt time.Time     `json:"updated_at" db:"updated_at"`
+	ID              uuid.UUID            `json:"id" db:"id" rw:"r"`
+	Code            ReServiceCode        `json:"code" db:"code" rw:"r"`
+	Priority        int                  `db:"priority" rw:"r"`
+	Name            string               `json:"name" db:"name" rw:"r"`
+	CreatedAt       time.Time            `json:"created_at" db:"created_at" rw:"r"`
+	UpdatedAt       time.Time            `json:"updated_at" db:"updated_at" rw:"r"`
+	ServiceLocation *ServiceLocationType `db:"service_location" rw:"r"`
```
Comment (danieljordan-caci) [inline pkg/models/re_service.go:153]: Can you add `ServiceLocation` above the `CreatedAt` line? Just to keep in line what we're doing in other models.

Comment (danieljordan-caci) [reply pkg/models/re_service.go:153]: Silly, I know.

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Couple of small changes - rest looks good.

File: pkg/models/re_service.go (hunk 2 above)
Comment (brianmanley-caci) [reply pkg/models/re_service.go]: Hey I'm all for consistency. This is fixed now.

File: pkg/models/re_service_item.go (hunk 1 above)
Comment (brianmanley-caci) [reply pkg/models/re_service_item.go]: I updated this to `r`. It looked like that was consistent with some of the other parameter names.

**Review by danieljordan-caci — APPROVED** [review]

**Pull request merged**